lib/api/v3/work_packages/work_package_list_helpers.rb

```diff
@@ -0,0 +1,92 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+module API
+  module V3
+    module WorkPackages
+      module WorkPackageListHelpers
+        extend Grape::API::Helpers
+
+        def work_packages_by_params(project: nil)
+          query = Query.new({ name: '_', project: project })
+          set_filters_from_json(query, params[:filters]) if params[:filters]
+          set_sorting_from_json(query, params[:sort_by]) if params[:sort_by]
+
+          collection_representer(query.results.sorted_work_packages,
+                                 project: project,
+                                 filter_json: params[:filters],
+                                 sort_json: params[:sort_by])
+        end
+
+        def set_filters_from_json(query, json)
+          filters = JSON.parse(json)
+          operators = filters.inject({}) { |result, filter|
+            attribute = filter.keys.first # there should only be one attribute per filter
+            result[attribute] = filter[attribute]['operator']
+            result
+          }
+          values = filters.inject({}) { |result, filter|
+            attribute = filter.keys.first # there should only be one attribute per filter
+            result[attribute] = filter[attribute]['values']
+            result
+          }
+
+          query.filters = []
+          query.add_filters(filters.map(&:keys).flatten, operators, values)
+        end
+
+        def set_sorting_from_json(query, json)
+          query.sort_criteria = JSON.parse(json)
+        end
+
+        def collection_representer(work_packages, project:, filter_json:, sort_json:)
```

Comment:
Assignment Branch Condition size for collection_representer is too high. [19.75/15]